webrecorder/browsertrix
# New create crawl config user workflow

Resolves #364.

### Manual testing
Install new shoelace version with `yarn`, then:
1. Run app with `yarn start`
2. Create new job config
    - Verify URL List flow functions as expected
    - Verify Seeded flow functions as expected
3. Edit newly created job config/crawl template. Verify it can still be edited.
4. Duplicate crawl template. Verify it takes you to new job config flow as URL list.

### Screenshots (in Firefox)
<img width="1139" alt="Screen Shot 2022-11-30 at 6 31 04 PM" src="https://user-images.githubusercontent.com/4672952/204951880-063f1efc-deb5-4080-8143-a7fa9da4ebe2.png">
<img width="900" alt="Screen Shot 2022-11-30 at 6 31 13 PM" src="https://user-images.githubusercontent.com/4672952/204951885-38905f75-fb77-496f-b4f6-57863088d95c.png">
<img width="897" alt="Screen Shot 2022-11-30 at 6 31 23 PM" src="https://user-images.githubusercontent.com/4672952/204951887-2ace5e2d-a1ea-472e-9323-5f69e8addecf.png">
<img width="896" alt="Screen Shot 2022-11-30 at 6 31 34 PM" src="https://user-images.githubusercontent.com/4672952/204951890-76249832-ff37-4786-91e4-fa25d3e61d6c.png">
<img width="903" alt="Screen Shot 2022-11-30 at 6 31 40 PM" src="https://user-images.githubusercontent.com/4672952/204951894-56c64306-9f00-4cdf-b882-605bd68932bd.png">

**Seeded crawl URL:**
<img width="895" alt="Screen Shot 2022-11-30 at 6 31 57 PM" src="https://user-images.githubusercontent.com/4672952/204951896-daf4dda1-5cab-4ba3-84a9-a731388bb75d.png">

**Seeded crawl URL validation:**
<img width="529" alt="Screen Shot 2022-11-30 at 6 32 11 PM" src="https://user-images.githubusercontent.com/4672952/204951897-93329119-20e9-4f63-be46-d4aac17ef104.png">

### Follow-ups
- [x] https://github.com/webrecorder/browsertrix-cloud/pull/392
- [x] https://github.com/webrecorder/browsertrix-cloud/pull/395
- [x] Specify job type (blocked by https://github.com/webrecorder/browsertrix-cloud/issues/363)
- [x] Improve duplicate config (only name and seed URLs copied right now, also blocked by https://github.com/webrecorder/browsertrix-cloud/issues/363)
- [ ] Handle smaller screen sizes: https://github.com/webrecorder/browsertrix-cloud/issues/408
